In [ ]:
## Bar chart race: https://app.flourish.studio/

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
!pip install pandas requests openpyxl

# Importar tabelas

In [ ]:
# @title Importando o Campeonato Brasileiro (LIG_A)
df_liga = pd.read_excel('https://github.com/adonisdario/data-science/raw/refs/heads/main/brasil-clubes-rank-2025/tabelas/Campeonato%20Brasileiro%20Rank.xlsx')
df_liga.index = df_liga.index + 1
df_liga = df_liga.iloc[:, 1:]
df_liga

Time  LIG_A_1  LIG_A_2  LIG_A_4  LIG_A_8  LIG_A_12  \
1              Palmeiras       12        5        9       15        11   
2                 Santos        8        8        4       16        14   
3            Corinthians        7        3       10       14         9   
4               Flamengo        7        3        6       13        14   
5              São Paulo        6        6        8       12        16   
..                   ...      ...      ...      ...      ...       ...   
158                Guará        0        0        0        0         0   
159          Operário-PR        0        0        0        0         0   
160  Potiguar de Mossoró        0        0        0        0         0   
161          Sport Belém        0        0        0        0         0   
162             Ubiratan        0        0        0        0         0   

     LIG_A_NR  LIG_A_16  LIG_A_24  LIG_A_32  LIG_A_64  LIG_A_100  LIG_A_R  \
1           3         1         2         1         0          0        2   
2           3         0         8         1         0          0        1   
3           7         2         1         2         0          0        1   
4           8         6         2         0         0          0        0   
5           3         2         3         1         0          0        0   
..        ...       ...       ...       ...       ...        ...      ...   
158         0         0         0         0         0          1        0   
159         0         0         0         0         0          1        0   
160         0         0         0         0         0          1        0   
161         0         0         0         0         0          1        0   
162         0         0         0         0         0          1        0   

     LIG_A_ULT  LIG_A_P      PTS  
1            0       61  2036.42  
2            2       63  1689.67  
3            2       56  1485.00  
4            2       59  1467.00  
5            0       57  1459.00  
..         ...      ...      ...  
158          1        1     1.00  
159          1        1     1.00  
160          1        1     1.00  
161          1        1     1.00  
162          1        1     1.00  

[162 rows x 16 columns]

In [ ]:
# @title Importando a Copa do Brasil (COP_A)
df_copa = pd.read_excel('https://github.com/adonisdario/data-science/raw/refs/heads/main/brasil-clubes-rank-2025/tabelas/Copa%20do%20Brasil%20Rank.xlsx')
df_copa.index = df_copa.index + 1
df_copa = df_copa.iloc[:, 1:]
df_copa

Time  COP_1  COP_2  COP_4  COP_8  COP_16  COP_F1  COP_F2  \
1           Flamengo      5      5      7      7       2       1       1   
2             Grêmio      5      4      7      5       6       2       1   
3           Cruzeiro      6      2      3      2       8       4       1   
4          Palmeiras      4      1      4      9       8       2       1   
5        Corinthians      3      4      3      7       9       2       0   
..               ...    ...    ...    ...    ...     ...     ...     ...   
375      Rondoniense      0      0      0      0       0       0       0   
376        São Bento      0      0      0      0       0       0       0   
377           Toledo      0      0      0      0       0       0       0   
378       Vilhenense      0      0      0      0       0       0       0   
379  Votuporanguense      0      0      0      0       0       0       0   

     COP_F3  COP_F4  COP_P  COP_BON         PTS  
1         0       0     28        5  808.866667  
2         1       0     31        5  779.588889  
3         1       0     27        6  764.811111  
4         0       0     29        4  571.477778  
5         0       0     28        5  555.222222  
..      ...     ...    ...      ...         ...  
375       0       1      1        0    0.170000  
376       0       1      1        0    0.170000  
377       0       1      1        0    0.170000  
378       0       1      1        0    0.170000  
379       0       1      1        0    0.170000  

[379 rows x 13 columns]

In [ ]:
# @title Importando a Libertadores (CON_A)
df_liberta = pd.read_excel('https://github.com/adonisdario/data-science/raw/refs/heads/main/brasil-clubes-rank-2025/tabelas/Libertadores%20Rank.xlsx')
df_liberta.index = df_liberta.index + 1
df_liberta = df_liberta.iloc[:, 1:]
df_liberta

Time  CON_A_1  CON_A_2  CON_A_SF  CON_A_QF  CON_A_OF  \
1              Palmeiras        3        3         5         3         6   
2              São Paulo        3        3         4         4         3   
3                 Grêmio        3        2         5         4         5   
4                 Santos        3        2         4         4         1   
5               Flamengo        3        1         2         4         5   
6               Cruzeiro        2        2         2         5         6   
7          Internacional        2        1         4         1         4   
8       Atlético Mineiro        1        1         2         3         4   
9            Corinthians        1        0         1         3         8   
10            Fluminense        1        1         0         4         1   
11              Botafogo        1        0         2         1         1   
12         Vasco da Gama        1        0         0         3         1   
13  Athletico Paranaense        0        2         0         0         5   
14           São Caetano        0        1         0         1         1   
15               Guarani        0        0         1         0         1   
16                 Bahia        0        0         0         2         0   
17              Criciúma        0        0         0         1         0   
18                 Sport        0        0         0         0         1   
19              Coritiba        0        0         0         0         0   
20             Fortaleza        0        0         0         0         1   
21                 Goiás        0        0         0         0         1   
22                Paraná        0        0         0         0         1   
23              Paysandu        0        0         0         0         1   
24           Chapecoense        0        0         0         0         0   
25   Red Bull Bragantino        0        0         0         0         0   
26       América Mineiro        0        0         0         0         0   
27                 Bangu        0        0         0         0         0   
28             Juventude        0        0         0         0         0   
29               Náutico        0        0         0         0         0   
30              Paulista        0        0         0         0         0   
31           Santo André        0        0         0         0         0   

    CON_A_1F  CON_A_PR  CON_A_P         PTS  
1          4         0       24  498.222222  
2          4         1       22  483.000000  
3          2         1       22  464.444444  
4          2         0       16  439.666667  
5          5         0       20  401.944444  
6          0         0       17  336.666667  
7          3         0       15  301.166667  
8          3         0       14  190.055556  
9          2         3       18  158.444444  
10         2         1       10  162.222222  
11         1         0        6  132.611111  
12         4         0        9  126.000000  
13         2         0        9   86.333333  
14         0         0        3   42.222222  
15         1         0        3   15.944444  
16         0         1        3   11.444444  
17         0         0        1    5.555556  
18         1         0        2    4.833333  
19         2         0        2    3.000000  
20         0         1        2    3.666667  
21         0         0        1    3.333333  
22         0         0        1    3.333333  
23         0         0        1    3.333333  
24         1         1        2    1.833333  
25         1         1        2    1.833333  
26         1         0        1    1.500000  
27         1         0        1    1.500000  
28         1         0        1    1.500000  
29         1         0        1    1.500000  
30         1         0        1    1.500000  
31         1         0        1    1.500000

In [ ]:
# @title Importando o Mundial de Clubes (MUN)
df_mundial = pd.read_excel('https://github.com/adonisdario/data-science/raw/refs/heads/main/brasil-clubes-rank-2025/tabelas/Mundial%20Rank.xlsx')
df_mundial.index = df_mundial.index + 1
df_mundial = df_mundial.iloc[:, 1:]
df_mundial

Time  MUN_A_1  MUN_A_2  MUN_A_3  MUN_A_4  MUN_A_8  MUN_B_1  \
1          São Paulo        1        0        0        0        0        2   
2             Santos        0        1        0        0        0        2   
3        Corinthians        2        0        0        0        0        0   
4             Grêmio        0        1        0        0        0        1   
5           Flamengo        0        1        1        0        0        1   
6          Palmeiras        0        1        0        1        0        0   
7      Internacional        1        0        1        0        0        0   
8      Vasco da Gama        0        1        0        0        0        0   
9           Cruzeiro        0        0        0        0        0        0   
10        Fluminense        0        1        0        0        0        0   
11  Atlético Mineiro        0        0        1        0        0        0   
12          Botafogo        0        0        0        0        1        0   

    MUN_B_2  MUN_C_1  MUN_C_3  MUN_P         PTS  
1         0        0        0      3  270.000000  
2         0        0        0      3  203.333333  
3         0        0        0      2  200.000000  
4         1        0        0      3  146.666667  
5         0        0        0      3  135.000000  
6         1        1        0      4  112.777778  
7         0        0        0      2  116.666667  
8         1        0        1      3   68.333333  
9         2        0        0      2   56.666667  
10        0        0        0      1   33.333333  
11        0        0        0      1   16.666667  
12        0        0        0      1    5.555556

In [ ]:
# @title Importando a Sulamericana (CON_B)
df_sula = pd.read_excel('https://github.com/adonisdario/data-science/raw/refs/heads/main/brasil-clubes-rank-2025/tabelas/Sulamericana%20Rank.xlsx')
df_sula.index = df_sula.index + 1
df_sula = df_sula.iloc[:, 1:]
df_sula

Time  CON_B_1  CON_B_2  CON_B_SF  CON_B_QF  CON_B_OF  \
1              São Paulo        2        1         3         2         1   
2       Atlético Mineiro        2        1         3         1         0   
3   Athletico Paranaense        2        0         1         2         1   
4               Botafogo        1        0         0         3         5   
5          Internacional        1        0         1         2         1   
6                 Santos        1        0         0         3         2   
7            Chapecoense        1        0         0         1         1   
8             Fluminense        0        1         1         3         4   
9            Corinthians        0        0         4         2         2   
10   Red Bull Bragantino        0        1         0         1         4   
11                 Goiás        0        1         0         0         4   
12              Flamengo        0        1         0         0         2   
13             Fortaleza        0        1         0         1         0   
14              Cruzeiro        0        1         0         0         1   
15           Ponte Preta        0        1         0         0         1   
16         Vasco da Gama        0        0         2         1         2   
17                   CSA        0        1         0         0         0   
18             Palmeiras        0        0         1         1         2   
19                 Bahia        0        0         0         2         2   
20                Grêmio        0        0         0         2         2   
21               Vitória        0        0         0         1         3   
22   Atlético Goianiense        0        0         1         0         1   
23                 Sport        0        0         0         1         2   
24        Sampaio Corrêa        0        0         1         0         0   
25                 Ceará        0        0         0         1         1   
26              Coritiba        0        0         0         1         1   
27                Paraná        0        0         0         1         0   
28       América Mineiro        0        0         0         1         0   
29                  Avaí        0        0         0         1         0   
30            Portuguesa        0        0         0         0         1   
31      América de Natal        0        0         0         0         1   
32              Brasília        0        0         0         0         1   
33               Guarani        0        0         0         0         1   
34            Santa Cruz        0        0         0         0         1   
35             Vila Nova        0        0         0         0         1   
36                Cuiabá        0        0         0         0         0   
37           Figueirense        0        0         0         0         0   
38              Criciúma        0        0         0         0         0   
39           São Caetano        0        0         0         0         0   
40             Joinville        0        0         0         0         0   
41             Juventude        0        0         0         0         0   
42               Náutico        0        0         0         0         0   

    CON_B_PO  CON_B_GP  CON_B_F1  CON_B_F2  CON_B_F3  CON_B_P     PTS  
1          0         0         5         1         0       15  282.76  
2          0         0         3         3         1       14  273.64  
3          0         0         3         0         0        9  226.46  
4          0         0         2         0         0       11  133.93  
5          1         0         1         1         0        8  127.31  
6          0         1         2         1         0       10  125.27  
7          0         0         0         1         0        4  109.04  
8          0         1         2         1         0       13   76.37  
9          0         1         1         1         0       11   63.85  
10         0         0         

In [ ]:
# @title Importando a Supercopa do Brasil (LIG_SC) e a Recopa Sulamericana (CON_SC)
df_sup_copa = pd.read_excel('https://github.com/adonisdario/data-science/raw/refs/heads/main/brasil-clubes-rank-2025/tabelas/Supercopas%20Rank.xlsx')
df_sup_copa.index = df_sup_copa.index + 1
df_sup_copa = df_sup_copa.iloc[:, 1:]
df_sup_copa

Time  CON_SC_1  LIG_SC_1  PTS
1           Flamengo         1         3  595
2             Grêmio         2         1  590
3          São Paulo         2         1  590
4      Internacional         2         0  470
5   Atlético Mineiro         1         1  355
6        Corinthians         1         1  355
7          Palmeiras         1         1  355
8           Cruzeiro         1         0  235
9         Fluminense         1         0  235
10            Santos         1         0  235

In [ ]:
# @title Importando a Copa dos Campeões Regionais (COP_B)
df_copa_camp_reg = pd.read_excel('https://github.com/adonisdario/data-science/raw/refs/heads/main/brasil-clubes-rank-2025/tabelas/Copa%20dos%20Campe%C3%B5es%20Rank.xlsx')
df_copa_camp_reg.index = df_copa_camp_reg.index + 1
df_copa_camp_reg = df_copa_camp_reg.iloc[:, 1:]
df_copa_camp_reg

Time  COP_B_1  COP_B_2  COP_B_4  COP_B_8  COP_B_16  \
1               Flamengo        1        0        2        0         0   
2              Palmeiras        1        0        1        0         0   
3               Paysandu        1        0        0        0         0   
4               Cruzeiro        0        1        1        1         0   
5              São Paulo        0        1        1        0         1   
6                  Sport        0        1        0        1         0   
7                  Goiás        0        0        0        2         1   
8               Coritiba        0        0        1        0         0   
9                  Bahia        0        0        0        2         0   
10               Vitória        0        0        0        2         0   
11           Corinthians        0        0        0        1         1   
12       São Raimundo-AM        0        0        0        1         1   
13       América Mineiro        0        0        0        1         0   
14            Fluminense        0        0        0        1         0   
15  Athletico Paranaense        0        0        0        0         1   
16      Atlético Mineiro        0        0        0        0         1   
17                Grêmio        0        0        0        0         1   
18               Náutico        0        0        0        0         1   
19           São Caetano        0        0        0        0         1   
20         Vasco da Gama        0        0        0        0         1   

    COP_B_P   COP_B_PTS  
1         3  122.222222  
2         2  111.111111  
3         1  100.000000  
4         3   50.000000  
5         3   47.777778  
6         2   38.888889  
7         3   14.444444  
8         1   11.111111  
9         2   11.111111  
10        2   11.111111  
11        2    8.888889  
12        2    8.888889  
13        1    5.555556  
14        1    5.555556  
15        1    3.333333  
16        1    3.333333  
17        1    3.333333  
18        1    3.333333  
19        1    3.333333  
20        1    3.333333

# Juntando as tabelas

In [ ]:
df1 = df_copa.merge(df_liga, on='Time', how='outer')
df1.fillna(0, inplace=True)
df1.rename(columns={'PTS_x': 'COP_PTS', 'PTS_y': 'LIG_A_PTS'}, inplace=True)
df1

Time  COP_1  COP_2  COP_4  COP_8  COP_16  COP_F1  COP_F2  \
0     15 de Novembro    0.0    0.0    1.0    0.0     1.0     0.0     0.0   
1         4 de Julho    0.0    0.0    0.0    0.0     0.0     3.0     2.0   
2                ABC    0.0    0.0    0.0    1.0     2.0    11.0     8.0   
3              ADESG    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
4                ASA    0.0    0.0    0.0    0.0     0.0     5.0     7.0   
..               ...    ...    ...    ...    ...     ...     ...     ...   
416      Ypiranga-AP    0.0    0.0    0.0    0.0     0.0     2.0     3.0   
417       Água Santa    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
418       Água Verde    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
419      Águia Negra    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
420  Águia de Marabá    0.0    0.0    0.0    0.0     0.0     3.0     1.0   

     COP_F3  COP_F4  ...  LIG_A_NR  LIG_A_16  LIG_A_24  LIG_A_32  LIG_A_64  \
0       0.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
1       1.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
2       1.0     1.0  ...       0.0       1.0       5.0       2.0       4.0   
3       0.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
4       4.0     2.0  ...       0.0       0.0       0.0       0.0       1.0   
..      ...     ...  ...       ...       ...       ...       ...       ...   
416     2.0     1.0  ...       0.0       0.0       0.0       0.0       0.0   
417     0.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
418     0.0     0.0  ...       0.0       0.0       1.0       0.0       0.0   
419     4.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
420     1.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   

     LIG_A_100  LIG_A_R  LIG_A_ULT  LIG_A_P  LIG_A_PTS  
0          0.0      0.0        0.0      0.0        0.0  
1          0.0      0.0        0.0      0.0        0.0  
2          0.0      0.0        7.0     14.0       74.0  
3          0.0      0.0        0.0      0.0        0.0  
4          0.0      0.0        0.0      1.0        2.0  
..         ...      ...        ...      ...        ...  
416        0.0      0.0        0.0      0.0        0.0  
417        0.0      0.0        0.0      0.0        0.0  
418        0.0      0.0        1.0      1.0        5.0  
419        0.0      0.0        0.0      0.0        0.0  
420        0.0      0.0        0.0      0.0        0.0  

[421 rows x 28 columns]

In [ ]:
df2 = df_liberta.merge(df_mundial, on='Time', how='outer')
df2.fillna(0, inplace=True)
df2.rename(columns={'PTS_x': 'CON_A_PTS', 'PTS_y': 'MUN_PTS'}, inplace=True)
df2

Time  CON_A_1  CON_A_2  CON_A_SF  CON_A_QF  CON_A_OF  \
0        América Mineiro        0        0         0         0         0   
1   Athletico Paranaense        0        2         0         0         5   
2       Atlético Mineiro        1        1         2         3         4   
3                  Bahia        0        0         0         2         0   
4                  Bangu        0        0         0         0         0   
5               Botafogo        1        0         2         1         1   
6            Chapecoense        0        0         0         0         0   
7            Corinthians        1        0         1         3         8   
8               Coritiba        0        0         0         0         0   
9               Criciúma        0        0         0         1         0   
10              Cruzeiro        2        2         2         5         6   
11              Flamengo        3        1         2         4         5   
12            Fluminense        1        1         0         4         1   
13             Fortaleza        0        0         0         0         1   
14                 Goiás        0        0         0         0         1   
15                Grêmio        3        2         5         4         5   
16               Guarani        0        0         1         0         1   
17         Internacional        2        1         4         1         4   
18             Juventude        0        0         0         0         0   
19               Náutico        0        0         0         0         0   
20             Palmeiras        3        3         5         3         6   
21                Paraná        0        0         0         0         1   
22              Paulista        0        0         0         0         0   
23              Paysandu        0        0         0         0         1   
24   Red Bull Bragantino        0        0         0         0         0   
25           Santo André        0        0         0         0         0   
26                Santos        3        2         4         4         1   
27                 Sport        0        0         0         0         1   
28           São Caetano        0        1         0         1         1   
29             São Paulo        3        3         4         4         3   
30         Vasco da Gama        1        0         0         3         1   

    CON_A_1F  CON_A_PR  CON_A_P   CON_A_PTS  ...  MUN_A_2  MUN_A_3  MUN_A_4  \
0          1         0        1    1.500000  ...      0.0      0.0      0.0   
1          2         0        9   86.333333  ...      0.0      0.0      0.0   
2          3         0       14  190.055556  ...      0.0      1.0      0.0   
3          0         1        3   11.444444  ...      0.0      0.0      0.0   
4          1         0        1    1.500000  ...      0.0      0.0      0.0   
5          1         0        6  132.611111  ...      0.0      0.0      0.0   
6          1         1        2    1.833333  ...      0.0      0.0      0.0   
7          2         3       18  158.444444  ...      0.0      0.0      0.0   
8          2         0        2    3.000000  ...      0.0      0.0      0.0   
9          0         0        1    5.555556  ...      0.0      0.0      0.0   
10         0         0       17  336.666667  ...      0.0      0.0      0.0   
11         5         0       20  401.944444  ...      1.0      1.0      0.0   
12         2         1       10  162.222222  ...      1.0      0.0      0.0   
13         0         1        2    3.666667  ...      0.0      0.0      0.0   
14         0         0        1    3.333333  ...      0.0      0.0      0.0   
15         2         1       22  464.444444  ...      1.0      0.0      0.0   
16         1         0        3   15.944444  ...      0.0      0.0      0.0   
17         3         0       15  301.166667  ...      0.0      1.0      0.0   
18         1         0        1    1.500000  ...      0.0      0.0      0.0   
19     

In [ ]:
df3 = df_sula.merge(df_sup_copa, on='Time', how='outer')
df3.fillna(0, inplace=True)
df3.rename(columns={'PTS_x': 'CON_B_PTS', 'PTS_y': 'SCOP_PTS'}, inplace=True)
df3

Time  CON_B_1  CON_B_2  CON_B_SF  CON_B_QF  CON_B_OF  \
0        América Mineiro        0        0         0         1         0   
1       América de Natal        0        0         0         0         1   
2   Athletico Paranaense        2        0         1         2         1   
3    Atlético Goianiense        0        0         1         0         1   
4       Atlético Mineiro        2        1         3         1         0   
5                   Avaí        0        0         0         1         0   
6                  Bahia        0        0         0         2         2   
7               Botafogo        1        0         0         3         5   
8               Brasília        0        0         0         0         1   
9                    CSA        0        1         0         0         0   
10                 Ceará        0        0         0         1         1   
11           Chapecoense        1        0         0         1         1   
12           Corinthians        0        0         4         2         2   
13              Coritiba        0        0         0         1         1   
14              Criciúma        0        0         0         0         0   
15              Cruzeiro        0        1         0         0         1   
16                Cuiabá        0        0         0         0         0   
17           Figueirense        0        0         0         0         0   
18              Flamengo        0        1         0         0         2   
19            Fluminense        0        1         1         3         4   
20             Fortaleza        0        1         0         1         0   
21                 Goiás        0        1         0         0         4   
22                Grêmio        0        0         0         2         2   
23               Guarani        0        0         0         0         1   
24         Internacional        1        0         1         2         1   
25             Joinville        0        0         0         0         0   
26             Juventude        0        0         0         0         0   
27               Náutico        0        0         0         0         0   
28             Palmeiras        0        0         1         1         2   
29                Paraná        0        0         0         1         0   
30           Ponte Preta        0        1         0         0         1   
31            Portuguesa        0        0         0         0         1   
32   Red Bull Bragantino        0        1         0         1         4   
33        Sampaio Corrêa        0        0         1         0         0   
34            Santa Cruz        0        0         0         0         1   
35                Santos        1        0         0         3         2   
36                 Sport        0        0         0         1         2   
37           São Caetano        0        0         0         0         0   
38             São Paulo        2        1         3         2         1   
39         Vasco da Gama        0        0         2         1         2   
40             Vila Nova        0        0         0         0         1   
41               Vitória        0        0         0         1         3   

    CON_B_PO  CON_B_GP  CON_B_F1  CON_B_F2  CON_B_F3  CON_B_P  CON_B_PTS  \
0          0         0         0         0         0        1       5.56   
1          0         0         0         0         0        1       3.33   
2          0         0         3         0         0        9     226.46   
3          0         1         0         0         0        3      15.62   
4          0         0         3         3         1       14     273.64   
5          0         0         0         0         0        1       5.56   
6          0         1         2         1         0        8      19.71   
7          0         0         2         0         0       11     133.93   
8          0         0         0         0         0        1      

In [ ]:
df4 = df1.merge(df2, on='Time', how='outer')
df4.fillna(0, inplace=True)
df4

Time  COP_1  COP_2  COP_4  COP_8  COP_16  COP_F1  COP_F2  \
0     15 de Novembro    0.0    0.0    1.0    0.0     1.0     0.0     0.0   
1         4 de Julho    0.0    0.0    0.0    0.0     0.0     3.0     2.0   
2                ABC    0.0    0.0    0.0    1.0     2.0    11.0     8.0   
3              ADESG    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
4                ASA    0.0    0.0    0.0    0.0     0.0     5.0     7.0   
..               ...    ...    ...    ...    ...     ...     ...     ...   
416      Ypiranga-AP    0.0    0.0    0.0    0.0     0.0     2.0     3.0   
417       Água Santa    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
418       Água Verde    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
419      Águia Negra    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
420  Águia de Marabá    0.0    0.0    0.0    0.0     0.0     3.0     1.0   

     COP_F3  COP_F4  ...  MUN_A_2  MUN_A_3  MUN_A_4  MUN_A_8  MUN_B_1  \
0       0.0     0.0  ...      0.0      0.0      0.0      0.0      0.0   
1       1.0     0.0  ...      0.0      0.0      0.0      0.0      0.0   
2       1.0     1.0  ...      0.0      0.0      0.0      0.0      0.0   
3       0.0     0.0  ...      0.0      0.0      0.0      0.0      0.0   
4       4.0     2.0  ...      0.0      0.0      0.0      0.0      0.0   
..      ...     ...  ...      ...      ...      ...      ...      ...   
416     2.0     1.0  ...      0.0      0.0      0.0      0.0      0.0   
417     0.0     0.0  ...      0.0      0.0      0.0      0.0      0.0   
418     0.0     0.0  ...      0.0      0.0      0.0      0.0      0.0   
419     4.0     0.0  ...      0.0      0.0      0.0      0.0      0.0   
420     1.0     0.0  ...      0.0      0.0      0.0      0.0      0.0   

     MUN_B_2  MUN_C_1  MUN_C_3  MUN_P  MUN_PTS  
0        0.0      0.0      0.0    0.0      0.0  
1        0.0      0.0      0.0    0.0      0.0  
2        0.0      0.0      0.0    0.0      0.0  
3        0.0      0.0      0.0    0.0      0.0  
4        0.0      0.0      0.0    0.0      0.0  
..       ...      ...      ...    ...      ...  
416      0.0      0.0      0.0    0.0      0.0  
417      0.0      0.0      0.0    0.0      0.0  
418      0.0      0.0      0.0    0.0      0.0  
419      0.0      0.0      0.0    0.0      0.0  
420      0.0      0.0      0.0    0.0      0.0  

[421 rows x 48 columns]

In [ ]:
df5 = df4.merge(df3, on='Time', how='outer')
df5.fillna(0, inplace=True)
df5

Time  COP_1  COP_2  COP_4  COP_8  COP_16  COP_F1  COP_F2  \
0     15 de Novembro    0.0    0.0    1.0    0.0     1.0     0.0     0.0   
1         4 de Julho    0.0    0.0    0.0    0.0     0.0     3.0     2.0   
2                ABC    0.0    0.0    0.0    1.0     2.0    11.0     8.0   
3              ADESG    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
4                ASA    0.0    0.0    0.0    0.0     0.0     5.0     7.0   
..               ...    ...    ...    ...    ...     ...     ...     ...   
416      Ypiranga-AP    0.0    0.0    0.0    0.0     0.0     2.0     3.0   
417       Água Santa    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
418       Água Verde    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
419      Águia Negra    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
420  Águia de Marabá    0.0    0.0    0.0    0.0     0.0     3.0     1.0   

     COP_F3  COP_F4  ...  CON_B_PO  CON_B_GP  CON_B_F1  CON_B_F2  CON_B_F3  \
0       0.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
1       1.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
2       1.0     1.0  ...       0.0       0.0       0.0       0.0       0.0   
3       0.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
4       4.0     2.0  ...       0.0       0.0       0.0       0.0       0.0   
..      ...     ...  ...       ...       ...       ...       ...       ...   
416     2.0     1.0  ...       0.0       0.0       0.0       0.0       0.0   
417     0.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
418     0.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
419     4.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   
420     1.0     0.0  ...       0.0       0.0       0.0       0.0       0.0   

     CON_B_P  CON_B_PTS  CON_SC_1  LIG_SC_1  SCOP_PTS  
0        0.0        0.0       0.0       0.0       0.0  
1        0.0        0.0       0.0       0.0       0.0  
2        0.0        0.0       0.0       0.0       0.0  
3        0.0        0.0       0.0       0.0       0.0  
4        0.0        0.0       0.0       0.0       0.0  
..       ...        ...       ...       ...       ...  
416      0.0        0.0       0.0       0.0       0.0  
417      0.0        0.0       0.0       0.0       0.0  
418      0.0        0.0       0.0       0.0       0.0  
419      0.0        0.0       0.0       0.0       0.0  
420      0.0        0.0       0.0       0.0       0.0  

[421 rows x 63 columns]

In [ ]:
df6 = df5.merge(df_copa_camp_reg, on='Time', how='outer')
df6.fillna(0, inplace=True)
df6

Time  COP_1  COP_2  COP_4  COP_8  COP_16  COP_F1  COP_F2  \
0     15 de Novembro    0.0    0.0    1.0    0.0     1.0     0.0     0.0   
1         4 de Julho    0.0    0.0    0.0    0.0     0.0     3.0     2.0   
2                ABC    0.0    0.0    0.0    1.0     2.0    11.0     8.0   
3              ADESG    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
4                ASA    0.0    0.0    0.0    0.0     0.0     5.0     7.0   
..               ...    ...    ...    ...    ...     ...     ...     ...   
416      Ypiranga-AP    0.0    0.0    0.0    0.0     0.0     2.0     3.0   
417       Água Santa    0.0    0.0    0.0    0.0     0.0     0.0     1.0   
418       Água Verde    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
419      Águia Negra    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
420  Águia de Marabá    0.0    0.0    0.0    0.0     0.0     3.0     1.0   

     COP_F3  COP_F4  ...  CON_SC_1  LIG_SC_1  SCOP_PTS  COP_B_1  COP_B_2  \
0       0.0     0.0  ...       0.0       0.0       0.0      0.0      0.0   
1       1.0     0.0  ...       0.0       0.0       0.0      0.0      0.0   
2       1.0     1.0  ...       0.0       0.0       0.0      0.0      0.0   
3       0.0     0.0  ...       0.0       0.0       0.0      0.0      0.0   
4       4.0     2.0  ...       0.0       0.0       0.0      0.0      0.0   
..      ...     ...  ...       ...       ...       ...      ...      ...   
416     2.0     1.0  ...       0.0       0.0       0.0      0.0      0.0   
417     0.0     0.0  ...       0.0       0.0       0.0      0.0      0.0   
418     0.0     0.0  ...       0.0       0.0       0.0      0.0      0.0   
419     4.0     0.0  ...       0.0       0.0       0.0      0.0      0.0   
420     1.0     0.0  ...       0.0       0.0       0.0      0.0      0.0   

     COP_B_4  COP_B_8  COP_B_16  COP_B_P  COP_B_PTS  
0        0.0      0.0       0.0      0.0        0.0  
1        0.0      0.0       0.0      0.0        0.0  
2        0.0      0.0       0.0      0.0        0.0  
3        0.0      0.0       0.0      0.0        0.0  
4        0.0      0.0       0.0      0.0        0.0  
..       ...      ...       ...      ...        ...  
416      0.0      0.0       0.0      0.0        0.0  
417      0.0      0.0       0.0      0.0        0.0  
418      0.0      0.0       0.0      0.0        0.0  
419      0.0      0.0       0.0      0.0        0.0  
420      0.0      0.0       0.0      0.0        0.0  

[421 rows x 70 columns]

# Pontuação

| Pontos | Competição | Pct % Total | Posição Copa | Pct % | Exclusividade | Posição Liga | Pct % |
|--|--|--|--|--|--|--|--|
||||COPAS|||LIGA|
|  15.000 | Mundial da FIFA     | **100,00%** | Vice       | 33,33% || Vice | 33,33%
|  12.750 | Intercontinental    |  **85,00%** | 3º         | 16,67% | Mundial FIFA | Top 4 | 23,75%
|   6.000 | 1o Mundial 1951     |  **40,00%** | Semis      | 11,11% || Top 8 | 17,50%
|   6.750 | Libertadores        |  **45,00%** | Quartas    |  5,56% || Top 12 | 12,50% |
|   2.250 | Série A             |  **15,00%** | Oitavas    |  3,33% || Top 16/N.Reb |  8,00%
|   2.100 | Copa do Brasil      |  **14,00%** | Playoff    |  2,42% | Sulamericana | Top 24/Reb | 5,00%
|   1.950 | Sulamericana        |  **13,00%** | -1F/Grupos |  1,50% || Top 32 |  4,00%
|   1.050 | Copa dos Campeões   |   **7,00%** | -2F        |  0,70% | Copa do Brasil | Top 64 |  2,00%
|  217,50 | Recopa Sulamericana |   **1,45%** | -3F/Prelim |  0,33% || Top 100 | 1,00%
|  112,50 | Supercopa Rei       |   **0,73%** | -4F/Pre 2  |  0,17% |
|         |                     |         | Pre 3      |  0,08% | Sulamericana

Obs.: Recopa e Supercopa não tem premiação para o vice

In [ ]:
pts_max = 15000
mundial_fifa = 100/100 * pts_max
intercontinental = 85/100 * pts_max
mundial_1951 = 40/100 * pts_max
libertadores = 45/100 * pts_max
serie_a = 15/100 * pts_max
copa_brasil = 14/100 * pts_max
sulamericana = 13/100 * pts_max
copa_campeoes = 7/100 * pts_max
recopa_sulamericana = 1.45/100 * pts_max
supercopa_rei = 0.73/100 * pts_max

In [ ]:
df_rank = df6.copy()
# Os pontos de cada Competição estão na escala de 0 a 100
df_rank['SCOP_PTS'] = (df_rank['CON_SC_1'] * recopa_sulamericana +
                      df_rank['LIG_SC_1'] * supercopa_rei)
df_rank['COP_PTS'] = (df_rank['COP_PTS']/100 * copa_brasil)
df_rank['LIG_A_PTS'] = (df_rank['LIG_A_PTS']/100 * serie_a)
df_rank['CON_A_PTS'] = (df_rank['CON_A_PTS']/100 * libertadores)
df_rank['MUN_PTS'] = (df_rank['MUN_PTS']/100 * mundial_fifa)
df_rank['CON_B_PTS'] = (df_rank['CON_B_PTS']/100 * sulamericana)
df_rank['COP_B_PTS'] = (df_rank['COP_B_PTS']/100 * copa_campeoes)

df_rank['PTS_TOTAL'] = round(df_rank['COP_PTS'] + df_rank['LIG_A_PTS'] +
                       df_rank['CON_A_PTS'] + df_rank['MUN_PTS'] +
                       df_rank['CON_B_PTS'] + df_rank['SCOP_PTS'] +
                       df_rank['COP_B_PTS'], 2)
df_rank
df_rank = df_rank.sort_values(by=['PTS_TOTAL', 'Time'], ascending=[False, True]).reset_index(drop=True)
df_rank.index += 1
df_rank

Time  COP_1  COP_2  COP_4  COP_8  COP_16  COP_F1  COP_F2  \
1          São Paulo    1.0    1.0    5.0    9.0     4.0     3.0     0.0   
2          Palmeiras    4.0    1.0    4.0    9.0     8.0     2.0     1.0   
3             Santos    1.0    1.0    3.0    5.0     6.0     5.0     0.0   
4           Flamengo    5.0    5.0    7.0    7.0     2.0     1.0     1.0   
5             Grêmio    5.0    4.0    7.0    5.0     6.0     2.0     1.0   
..               ...    ...    ...    ...    ...     ...     ...     ...   
417             Novo    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
418      Rondoniense    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
419           Toledo    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
420       Vilhenense    0.0    0.0    0.0    0.0     0.0     0.0     0.0   
421  Votuporanguense    0.0    0.0    0.0    0.0     0.0     0.0     0.0   

     COP_F3  COP_F4  ...  LIG_SC_1  SCOP_PTS  COP_B_1  COP_B_2  COP_B_4  \
1       0.0     0.0  ...       1.0     544.5      0.0      1.0      1.0   
2       0.0     0.0  ...       1.0     327.0      1.0      0.0      1.0   
3       0.0     0.0  ...       0.0     217.5      0.0      0.0      0.0   
4       0.0     0.0  ...       3.0     546.0      1.0      0.0      2.0   
5       1.0     0.0  ...       1.0     544.5      0.0      0.0      0.0   
..      ...     ...  ...       ...       ...      ...      ...      ...   
417     0.0     1.0  ...       0.0       0.0      0.0      0.0      0.0   
418     0.0     1.0  ...       0.0       0.0      0.0      0.0      0.0   
419     0.0     1.0  ...       0.0       0.0      0.0      0.0      0.0   
420     0.0     1.0  ...       0.0       0.0      0.0      0.0      0.0   
421     0.0     1.0  ...       0.0       0.0      0.0      0.0      0.0   

     COP_B_8  COP_B_16  COP_B_P    COP_B_PTS  PTS_TOTAL  
1        0.0       1.0      3.0   501.666667  118371.15  
2        0.0       0.0      2.0  1166.666667  110324.53  
3        0.0       0.0      0.0     0.000000  106006.17  
4        0.0       0.0      3.0  1283.333333   99995.79  
5        0.0       1.0      1.0    35.000000   98741.30  
..       ...       ...      ...          ...        ...  
417      0.0       0.0      0.0     0.000000       3.57  
418      0.0       0.0      0.0     0.000000       3.57  
419      0.0       0.0      0.0     0.000000       3.57  
420      0.0       0.0      0.0     0.000000       3.57  
421      0.0       0.0      0.0     0.000000       3.57  

[421 rows x 71 columns]

# Top 50 do ranking

In [ ]:
last_column = df_rank.columns[-1]
df_50 = df_rank[['Time', last_column]]
df_50.head(50)

Time  PTS_TOTAL
1              São Paulo  118371.15
2              Palmeiras  110324.53
3                 Santos  106006.17
4               Flamengo   99995.79
5                 Grêmio   98741.30
6            Corinthians   87432.58
7               Cruzeiro   78262.50
8          Internacional   75061.45
9       Atlético Mineiro   58734.60
10         Vasco da Gama   50592.75
11            Fluminense   48904.76
12              Botafogo   38119.36
13  Athletico Paranaense   29742.68
14                 Bahia   19808.83
15                 Sport   16753.27
16              Coritiba   15221.38
17                 Goiás   13655.19
18               Vitória   11750.60
19               Guarani   11615.23
20            Portuguesa    8937.26
21             Fortaleza    8815.64
22               Náutico    8521.95
23           Ponte Preta    7263.02
24                 Ceará    7260.03
25             Juventude    6599.80
26           São Caetano    6226.21
27              Paysandu    6035.27
28   Red Bull Bragantino    6020.84
29            Santa Cruz    5486.06
30              Criciúma    5408.99
31                Paraná    4966.25
32           Figueirense    4248.12
33            America-RJ    4200.90
34           Chapecoense    4166.80
35       América Mineiro    3887.72
36                  Remo    3621.19
37   Atlético Goianiense    3441.21
38                   CSA    3373.51
39           Santo André    2826.97
40         Rio Branco-ES    2490.18
41                 Bangu    2396.39
42                   ABC    2396.34
43           Operário-MS    2389.45
44              Paulista    2317.45
45      América de Natal    2176.04
46                  Avaí    2068.27
47            Campinense    1922.51
48        Sampaio Corrêa    1737.72
49           Nacional-AM    1729.87
50              Metropol    1631.25

In [ ]:
# Exportar para Excel
df_rank.to_excel('Rank Histórico 2025.xlsx')

# Faixas de pontuação - Sturges

As faixas de pontuação serão criadas através da regra de Sturges

In [ ]:
last_column = df_rank.columns[-1]
df_pts = df_rank[['Time', last_column]]

In [ ]:
n = len(df_pts)
k = 1 + (10/3) * np.log10(n)
k = int(k.round(0))
frequencia = pd.value_counts(
    pd.cut(x = df_pts['PTS_TOTAL'], bins = k, include_lowest = True),
    sort = False
)
percentual = (pd.value_counts(
    pd.cut(x = df_pts['PTS_TOTAL'], bins = k, include_lowest = True),
    sort = False,
    normalize = True
    ) * 100
).round(2)
faixa_pts = pd.DataFrame({'Frequência': frequencia, 'Porcentagem (%)': percentual})
faixa_pts = faixa_pts.sort_index(ascending=False)
faixa_pts

/tmp/ipython-input-23-1150778850.py:4: FutureWarning: pandas.value_counts is deprecated and will be removed in a future version. Use pd.Series(obj).value_counts() instead.
  frequencia = pd.value_counts(
/tmp/ipython-input-23-1150778850.py:8: FutureWarning: pandas.value_counts is deprecated and will be removed in a future version. Use pd.Series(obj).value_counts() instead.
  percentual = (pd.value_counts(


Frequência  Porcentagem (%)
PTS_TOTAL                                           
(106534.392, 118371.15]           2             0.48
(94697.634, 106534.392]           3             0.71
(82860.876, 94697.634]            1             0.24
(71024.118, 82860.876]            2             0.48
(59187.36, 71024.118]             0             0.00
(47350.602, 59187.36]             3             0.71
(35513.844, 47350.602]            1             0.24
(23677.086, 35513.844]            1             0.24
(11840.328, 23677.086]            4             0.95
(-114.799, 11840.328]           404            95.96

Diferença de 11.836,76 entre prateleiras

**A -> Acima de 106.534,39** - São Paulo e Palmeiras

**B** -> Acima de 94.697,63 - Santos, Flamengo e Grêmio

**C** -> Acima de 82.860,87 - Corinthians

**D** -> Acima de 71.024,11 - Cruzeiro e Internacional

**E** -> Acima de 59.187,36 - ***vazio***

**F** -> Acima de 47.350,60 - Atlético Mineiro, Vasco da Gama e Fluminense

**G** -> Acima de 35.513,84 - Botafogo

**H** -> Acima de 23.677,08 - Athletico Paranaense

**I** -> Acima de 11.840,32 - Bahia, Sport, Coritiba e Goiás

**J** -> Até 11.840,32 - 404 times

## Sturges parte 2

In [ ]:
df_pts_pt2 = df_pts[df_pts['PTS_TOTAL'] <= 11840.328]
df_pts_pt2.head(50)

Time  PTS_TOTAL
18                 Vitória   11750.60
19                 Guarani   11615.23
20              Portuguesa    8937.26
21               Fortaleza    8815.64
22                 Náutico    8521.95
23             Ponte Preta    7263.02
24                   Ceará    7260.03
25               Juventude    6599.80
26             São Caetano    6226.21
27                Paysandu    6035.27
28     Red Bull Bragantino    6020.84
29              Santa Cruz    5486.06
30                Criciúma    5408.99
31                  Paraná    4966.25
32             Figueirense    4248.12
33              America-RJ    4200.90
34             Chapecoense    4166.80
35         América Mineiro    3887.72
36                    Remo    3621.19
37     Atlético Goianiense    3441.21
38                     CSA    3373.51
39             Santo André    2826.97
40           Rio Branco-ES    2490.18
41                   Bangu    2396.39
42                     ABC    2396.34
43             Operário-MS    2389.45
44                Paulista    2317.45
45        América de Natal    2176.04
46                    Avaí    2068.27
47              Campinense    1922.51
48          Sampaio Corrêa    1737.72
49             Nacional-AM    1729.87
50                Metropol    1631.25
51                Londrina    1590.98
52               Moto Club    1590.24
53  Desportiva Ferroviária    1528.67
54               Vila Nova    1526.98
55                     CRB    1525.12
56             Brasiliense    1396.04
57               Joinville    1370.67
58                 Sergipe    1257.74
59               Confiança    1199.07
60                   Treze    1182.39
61                  Cuiabá    1140.93
62                    Gama    1074.67
63               Americano    1017.12
64             Botafogo-SP     969.10
65                Goytacaz     945.00
66       Brasil de Pelotas     863.08
67        Inter de Limeira     850.75

In [ ]:
n = len(df_pts_pt2)
k = 1 + (10/3) * np.log10(n)
k = int(k.round(0))
frequencia = pd.value_counts(
    pd.cut(x = df_pts_pt2['PTS_TOTAL'], bins = k, include_lowest = True),
    sort = False
)
percentual = (pd.value_counts(
    pd.cut(x = df_pts_pt2['PTS_TOTAL'], bins = k, include_lowest = True),
    sort = False,
    normalize = True
    ) * 100
).round(2)
faixa_pts = pd.DataFrame({'Frequência': frequencia, 'Porcentagem (%)': percentual})
faixa_pts = faixa_pts.sort_index(ascending=False)
faixa_pts

/tmp/ipython-input-25-760342391.py:4: FutureWarning: pandas.value_counts is deprecated and will be removed in a future version. Use pd.Series(obj).value_counts() instead.
  frequencia = pd.value_counts(
/tmp/ipython-input-25-760342391.py:8: FutureWarning: pandas.value_counts is deprecated and will be removed in a future version. Use pd.Series(obj).value_counts() instead.
  percentual = (pd.value_counts(


Frequência  Porcentagem (%)
PTS_TOTAL                                                  
(10575.897, 11750.6]                     2             0.50
(9401.194, 10575.897]                    0             0.00
(8226.491, 9401.194]                     3             0.74
(7051.788, 8226.491]                     2             0.50
(5877.085, 7051.788]                     4             0.99
(4702.382, 5877.085]                     3             0.74
(3527.679, 4702.382]                     5             1.24
(2352.976, 3527.679]                     7             1.73
(1178.273, 2352.976]                    17             4.21
(-8.177999999999999, 1178.273]         361            89.36

Diferença de 1.264,43 entre as prateleiras I-J e 1.174,70 entre prateleiras J-S

**J -> Acima de 10.575,89** - Vitória e Guarani

**K** -> Acima de 9.401,19 - *vazio*

**L** -> Acima de 8.226,49 - Portuguesa, Fortaleza e Náutico

**M** -> Acima de 7.051,78 - Ponte Preta e Ceará

**N** -> Acima de 5.877,08 - Juventude, São Caetano, Paysandu e RB Bragantino

**O** -> Acima de 4.702,38 - Santa Cruz, Criciúma e Paraná

**P** -> Acima de 3.527,67 - Figueirense, America-RJ, Chapecoense, América-MG e Remo

**Q** -> Acima de 2.352,97 - Atlético-GO, CSA, Santo André, Rio Branco-ES, Bangou, ABC e Operário-MS

**R** -> Acima de 1.178,27 - 17 times

**S** -> Até 1.178,27 - 364 times

# Faixas de Pontuação - Friedman-Diaconis

In [ ]:
import matplotlib.pyplot as plt
df_fd = df_pts.copy()

# Cálculo do método Freedman-Diaconis
q75, q25 = np.percentile(df_fd['PTS_TOTAL'], [75 ,25])
iqr = q75 - q25

n = len(df_fd)
bin_width = 2 * iqr / (n ** (1/3))

# Número de bins (arredondado)
data_range = df_fd['PTS_TOTAL'].max() - df_fd['PTS_TOTAL'].min()
num_bins = int(np.ceil(data_range / bin_width))

# Corte dos dados em prateleiras
df_fd['Prateleira'] = pd.cut(df_fd['PTS_TOTAL'], bins=num_bins, labels=False)
df_fd['Prateleira'] = num_bins - df_fd['Prateleira']

# Exibir resultado
print(df_fd.head())
print(f"Número de prateleiras: {num_bins}")

df_fd

        Time  PTS_TOTAL  Prateleira
1  São Paulo  118371.15           1
2  Palmeiras  110324.53          95
3     Santos  106006.17         146
4   Flamengo   99995.79         217
5     Grêmio   98741.30         232
Número de prateleiras: 1394


Time  PTS_TOTAL  Prateleira
1          São Paulo  118371.15           1
2          Palmeiras  110324.53          95
3             Santos  106006.17         146
4           Flamengo   99995.79         217
5             Grêmio   98741.30         232
..               ...        ...         ...
417             Novo       3.57        1394
418      Rondoniense       3.57        1394
419           Toledo       3.57        1394
420       Vilhenense       3.57        1394
421  Votuporanguense       3.57        1394

[421 rows x 3 columns]

In [ ]:
df_fd.head(50)

Time  PTS_TOTAL  Prateleira
1              São Paulo  118371.15           1
2              Palmeiras  110324.53          95
3                 Santos  106006.17         146
4               Flamengo   99995.79         217
5                 Grêmio   98741.30         232
6            Corinthians   87432.58         365
7               Cruzeiro   78262.50         473
8          Internacional   75061.45         511
9       Atlético Mineiro   58734.60         703
10         Vasco da Gama   50592.75         799
11            Fluminense   48904.76         819
12              Botafogo   38119.36         946
13  Athletico Paranaense   29742.68        1044
14                 Bahia   19808.83        1161
15                 Sport   16753.27        1197
16              Coritiba   15221.38        1215
17                 Goiás   13655.19        1234
18               Vitória   11750.60        1256
19               Guarani   11615.23        1258
20            Portuguesa    8937.26        1289
21             Fortaleza    8815.64        1291
22               Náutico    8521.95        1294
23           Ponte Preta    7263.02        1309
24                 Ceará    7260.03        1309
25             Juventude    6599.80        1317
26           São Caetano    6226.21        1321
27              Paysandu    6035.27        1323
28   Red Bull Bragantino    6020.84        1324
29            Santa Cruz    5486.06        1330
30              Criciúma    5408.99        1331
31                Paraná    4966.25        1336
32           Figueirense    4248.12        1345
33            America-RJ    4200.90        1345
34           Chapecoense    4166.80        1345
35       América Mineiro    3887.72        1349
36                  Remo    3621.19        1352
37   Atlético Goianiense    3441.21        1354
38                   CSA    3373.51        1355
39           Santo André    2826.97        1361
40         Rio Branco-ES    2490.18        1365
41                 Bangu    2396.39        1366
42                   ABC    2396.34        1366
43           Operário-MS    2389.45        1366
44              Paulista    2317.45        1367
45      América de Natal    2176.04        1369
46                  Avaí    2068.27        1370
47            Campinense    1922.51        1372
48        Sampaio Corrêa    1737.72        1374
49           Nacional-AM    1729.87        1374
50              Metropol    1631.25        1375

In [ ]:
len(df_fd['Prateleira'].value_counts())

62

In [ ]:
df_fd['Prateleira'].value_counts().sort_index(ascending=True)

Prateleira
1         1
95        1
146       1
217       1
232       1
       ... 
1390     13
1391     11
1392     22
1393     44
1394    240
Name: count, Length: 62, dtype: int64

# Faixas de Pontuação - Scott

In [ ]:
# Copiar o DataFrame original
df_scott = df_pts.copy()

# Calcular largura do bin com a Regra de Scott
n = len(df_scott)
std_dev = df_scott['PTS_TOTAL'].std() # Desvio padrão
bin_width = 3.5 * std_dev / (n ** (1/3))

# Criar os bins com base na largura calculada
min_val = df_scott['PTS_TOTAL'].min()
max_val = df_scott['PTS_TOTAL'].max()
bins = np.arange(min_val, max_val + bin_width, bin_width)

# Aplicar os bins (com rótulos temporários)
df_scott['prateleira_temp'] = pd.cut(df_scott['PTS_TOTAL'], bins=bins, include_lowest=True)

# Mapear os intervalos para números contínuos, começando em 1
intervalos_unicos = df_scott['prateleira_temp'].unique()
mapa_intervalos = {intervalo: i+1 for i, intervalo in enumerate(sorted(intervalos_unicos))}
df_scott['prateleira_scott'] = (df_scott['prateleira_temp'].map(mapa_intervalos)).astype(int)
df_scott['prateleira_scott'] = len(df_scott['prateleira_scott'].value_counts()) - df_scott['prateleira_scott'] + 1

# Remover a coluna temporária
df_scott.drop(columns='prateleira_temp', inplace=True)

# Aplicar os bins ao DataFrame
df_scott['prateleira_scott_ori'] = pd.cut(df_scott['PTS_TOTAL'], bins=bins, labels=False, include_lowest=True)
df_scott['prateleira_scott_ori'] = df_scott['prateleira_scott_ori'].max() - df_scott['prateleira_scott_ori'] + 1

df_scott[df_scott['prateleira_scott'] < 11]

Time  PTS_TOTAL  prateleira_scott  prateleira_scott_ori
1              São Paulo  118371.15                 1                     1
2              Palmeiras  110324.53                 2                     2
3                 Santos  106006.17                 3                     3
4               Flamengo   99995.79                 3                     3
5                 Grêmio   98741.30                 4                     4
6            Corinthians   87432.58                 5                     5
7               Cruzeiro   78262.50                 6                     7
8          Internacional   75061.45                 6                     7
9       Atlético Mineiro   58734.60                 7                    10
10         Vasco da Gama   50592.75                 8                    11
11            Fluminense   48904.76                 8                    11
12              Botafogo   38119.36                 9                    13
13  Athletico Paranaense   29742.68                10                    14

## Scott parte 2

In [ ]:
# Copiar o DataFrame original
df_scott_2 = df_scott[17:].copy()

# Calcular largura do bin com a Regra de Scott
n = len(df_scott_2)
std_dev = df_scott_2['PTS_TOTAL'].std() # Desvio padrão
bin_width = 3.5 * std_dev / (n ** (1/3))

# Criar os bins com base na largura calculada
min_val = df_scott_2['PTS_TOTAL'].min()
max_val = df_scott_2['PTS_TOTAL'].max()
bins = np.arange(min_val, max_val + bin_width, bin_width)

# Aplicar os bins (com rótulos temporários)
df_scott_2['prateleira_temp'] = pd.cut(df_scott_2['PTS_TOTAL'], bins=bins, include_lowest=True)

# Mapear os intervalos para números contínuos, começando em 1
intervalos_unicos = df_scott_2['prateleira_temp'].unique()
mapa_intervalos = {intervalo: i+1 for i, intervalo in enumerate(sorted(intervalos_unicos))}
df_scott_2['prateleira_scott'] = (df_scott_2['prateleira_temp'].map(mapa_intervalos)).astype(int)
df_scott_2['prateleira_scott'] = len(df_scott_2['prateleira_scott'].value_counts()) - df_scott_2['prateleira_scott'] + 11

# Remover a coluna temporária
df_scott_2.drop(columns='prateleira_temp', inplace=True)

# Aplicar os bins ao DataFrame
df_scott_2['prateleira_scott_ori'] = pd.cut(df_scott_2['PTS_TOTAL'], bins=bins, labels=False, include_lowest=True)
df_scott_2['prateleira_scott_ori'] = df_scott_2['prateleira_scott_ori'].max() - df_scott_2['prateleira_scott_ori'] + 11

df_scott_2.head(50)

Time  PTS_TOTAL  prateleira_scott  prateleira_scott_ori
18                 Vitória   11750.60                11                    11
19                 Guarani   11615.23                11                    11
20              Portuguesa    8937.26                12                    15
21               Fortaleza    8815.64                12                    15
22                 Náutico    8521.95                13                    16
23             Ponte Preta    7263.02                14                    17
24                   Ceará    7260.03                14                    17
25               Juventude    6599.80                15                    18
26             São Caetano    6226.21                16                    19
27                Paysandu    6035.27                16                    19
28     Red Bull Bragantino    6020.84                16                    19
29              Santa Cruz    5486.06                17                    20
30                Criciúma    5408.99                17                    20
31                  Paraná    4966.25                18                    21
32             Figueirense    4248.12                19                    22
33              America-RJ    4200.90                19                    22
34             Chapecoense    4166.80                19                    22
35         América Mineiro    3887.72                19                    22
36                    Remo    3621.19                19                    22
37     Atlético Goianiense    3441.21                20                    23
38                     CSA    3373.51                20                    23
39             Santo André    2826.97                21                    24
40           Rio Branco-ES    2490.18                21                    24
41                   Bangu    2396.39                21                    24
42                     ABC    2396.34                21                    24
43             Operário-MS    2389.45                21                    24
44                Paulista    2317.45                21                    24
45        América de Natal    2176.04                21                    24
46                    Avaí    2068.27                22                    25
47              Campinense    1922.51                22                    25
48          Sampaio Corrêa    1737.72                22                    25
49             Nacional-AM    1729.87                22                    25
50                Metropol    1631.25                22                    25
51                Londrina    1590.98                22                    25
52               Moto Club    1590.24                22                    25
53  Desportiva Ferroviária    1528.67                22                    25
54               Vila Nova    1526.98                22                    25
55                     CRB    1525.12                22                    25
56             Brasiliense    1396.04                23                    26
57               Joinville    1370.67                23                    26
58                 Sergipe    1257.74                23                    26
59               Confiança    1199.07                23                    26
60                   Treze    1182.39                23                    26
61                  Cuiabá    1140.93                23                    26
62                    Gama    1074.67                23                    26
63               Americano    1017.12                23                    26
64             Botafogo-SP     969.10                23                    26
65                Goytacaz     945.00                23                    26
66       Brasil de Pelotas     863.08                23                    26
67        Inter de Limeira     850.75                23                    26

In [ ]:
len(df_scott_2['prateleira_scott_ori'].value_counts())

14

In [ ]:
df_scott['prateleira_scott'].value_counts().sort_index(ascending=True)

prateleira_scott
1       1
2       1
3       2
4       1
5       1
6       2
7       1
8       2
9       1
10      1
11      4
12      7
13    397
Name: count, dtype: int64

In [ ]:
df_scott_2['prateleira_scott'].value_counts().sort_index(ascending=True)

prateleira_scott
11      2
12      2
13      1
14      2
15      1
16      3
17      2
18      1
19      5
20      2
21      7
22     10
23     17
24    349
Name: count, dtype: int64